src/opera/util/input_validation.py

```diff
@@ -130,6 +131,133 @@ def validate_slc_s1_inputs(runconfig, logger, name):
             logger.critical(name, ErrorCode.INVALID_INPUT, error_msg)
 
 
+def get_burst_id_set(input_file_group, logger, name):
+    """
+    Compiles a set of burst_ids from a list of files defined in the runconfig file.
+    Each file in the list should have have a burst_id in the file name.
+
+    Parameters
+    ----------
+    input_file_group : list
+        List of files containing a burst_id
+    logger : PgeLogger
+        Logger passed by the calling PGE
+    name : str
+        PGE name
+
+    Returns
+    -------
+    burst_ids :set
+
+    Raises
+    ------
+    RuntimeError
+        If a burst_id is improperly formatted.
+
+    """
+    burst_ids = set()
+    for i in input_file_group:
+        reg_ex = r't\w{3}_\d{6}_iw[1|2|3]'
+        match = re.search(reg_ex, i)
+        if match:
+            burst_ids.add(re.findall(reg_ex, i)[0])
+        else:
+            msg = f'Input file present without properly formatted burst_id: {i}'
+            logger.critical(name, ErrorCode.INVALID_INPUT, msg)
+
+    return burst_ids
+
+
+def check_amp_mean_geometry_burst_ids(cslc_input_burst_ids, ancillary_file_list, logger, name):
```

Comment:
I love that this function is written at the abstraction level of "ancillary files", such that it will work for any arbitrary list of input ancillaries that contain burst IDs. Because of this, I think the name of the function should be changed to match, something like `check_disp_s1_ancillary_burst_ids`